setup_go.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -ex
+
+# Install gimme to get go1.15.11
+curl -sL -o /bin/gimme https://raw.githubusercontent.com/travis-ci/gimme/master/gimme
+chmod +x /bin/gimme
+eval $(gimme 1.15.11)
+
+# Install gcc 4.9 (needed to compile go, as it introduces -fdiagnostics-color which is required
+# in the post-compilation steps of go)
+# We don't add it to the PATH so that it's not used in the Agent / rtloader builds
+apt-get update && apt-get install -y gcc-4.9-backport
+
+git clone https://go.googlesource.com/go goroot && cd goroot
+git checkout go$GO_VERSION
+cd src 
+
+# Use gcc 4.9 + go1.15.11 to build the target go version
+CC=/usr/lib/gcc-4.9-backport/bin/gcc ./all.bash
```

Comment:
`make.bash` has these two variables. If we don't set `CC_FOR_TARGET`, does that mean that we use `gcc` 4.9 for cgo?

```
# CC: Command line to run to compile C code for GOHOSTARCH.
# Default is "gcc". Also supported: "clang".
#
# CC_FOR_TARGET: Command line to run to compile C code for GOARCH.
# This is used by cgo.  Default is CC.
```

Reply:
The only thing I could find is old docs from when the `CC_FOR_TARGET` option was introduced, but I believe the nuance is that `CC_FOR_TARGET` is used by `cgo` _when cross-compiling_: https://github.com/golang/go/commit/88d07b2cbcf0ad5e2e9936bcaec4fd25e38bec10, whereas `CC` is used for `cgo` when not cross-compiling (as the `for GOHOSTARCH` / `for GOARCH` distinction seems to show).

Reply:
I see. So if I understood correctly, the Go compiler uses cgo and these variables refer to the C compiler that you use for cgo when compiling/cross-compiling the Go compiler, but not to the one used by the Go compiler, which is independent, right?

Reply:
Yes, I believe that's it. To verify that this is correct, I'll modify the script to remove the `gcc 4.9` compiler after building go.

Reply:
Yes, I believe that's it. To verify that this is correct, I'll modify the script to remove the `gcc 4.9` compiler after building go.